v2/devdoc/log_sink_etw_requirements.md

```diff
@@ -0,0 +1,179 @@
+# `log_sink_etw` requirements
+
+`log_sink_etw` implements a log sink interface that logs the events as ETW events.
+
+## References
+
+[Event Tracing For Windows](https://learn.microsoft.com/en-us/windows-hardware/drivers/devtest/event-tracing-for-windows--etw-)
+
+## Exposed API
+
+```
+    extern const LOG_SINK_IF log_sink_etw;
+```
+
+### log_sink_etw.log_sink_log
+
+The signature of `log_sink_etw.log_sink_log` is:
+
+```c
+typedef void (*LOG_SINK_LOG_FUNC)(LOG_LEVEL log_level, LOG_CONTEXT_HANDLE log_context, const char* file, const char* func, int line, const char* message_format, ...);
+```
+
+`log_sink_etw.log_sink_log` implements logging to an ETW provider.
+
+If `message_format` is `NULL`, `log_sink_etw.log_sink_log` shall fail and return.
+
+`log_sink_etw_log` shall register the ETW TraceLogging provider by calling `TraceLoggingRegister` if it was not already done.
```

Comment:
>if it was not already done. [](http://example.com/codeflow?start=98&length=27)

404: details. Is this done with lazy_init? call_once? etc.

Also what's the params to TraceLoggingRegister? Would I be able to write UTs with this description?

Reply:
Added details.
For TraceLoggingRegister I'll have to do a bit of reverse engineering of the macro unfortunately and add that in too.